presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -246,6 +247,11 @@
      */
     void revokeTablePrivileges(Session session, QualifiedObjectName tableName, Set<Privilege> privileges, String grantee, boolean grantOption);
 
+    /**
+     * Gets the privileges for the specified table available to the given grantee
+     */
+    List<GrantInfo> listTablePrivileges(Session session, QualifiedTablePrefix prefix, String grantee);
```

Comment:
Pull the SPI changes for Metadata and ConnectorMetadata into a separate commit, please.